pelutils/test/test_pelutils.py

```diff
@@ -0,0 +1,39 @@
+import numpy as np
+
+from pelutils.ds import unique
+
+def test_unique():
+    # Simple case: Ordered numbers from 0 to 99
+    n = 100
+    a = np.arange(n, dtype=np.uint32)
+    u, index, inverse, counts = unique(a, return_index=True, return_inverse=True, return_counts=True)
+    assert np.all(a == u)
+    assert np.all(a == index)
+    assert np.all(a == inverse)
+    assert np.all(counts == 1)
+
+    # Slightly more complex case with some non-unique values
+    a[2:4] = 50
+    a[[5, 16, 3]] = 69
+    a = a.astype(np.float16)
+    u, index, inverse, counts = unique(a, return_index=True, return_inverse=True, return_counts=True)
+    argsort = np.argsort(u)
+    npu, npindex, npcounts = np.unique(a, return_index=True, return_counts=True)
+    assert np.all(u[argsort] == npu)
+    assert np.all(index[argsort] == npindex)
+    assert np.all(a == u[inverse])
+    assert np.all(counts[argsort] == npcounts)
+
+    # Multidimensional array
+    a = np.random.randint(0, 10, (10, 5, 5))
```

Comment:
Modigt at lave randomiseret testing på denne måde ;) Synes, der skal være noget seedsætteri inden, så man ikke får for mange flaky-fejl

Reply:
Løst ;)